internal/pkg/cli/deploy.go

```diff
@@ -186,45 +196,196 @@ func (o *deployOpts) maybeInitWkld() error {
+		for k := range prioritiesMap {
+			specifiedWorkloadList = append(specifiedWorkloadList, k)
+		}
+
+		if o.yesInitWkld != nil && !aws.BoolValue(o.yesInitWkld) {
+			// --all and --init-wkld=false: get only get initialized local workloads.
+			initializedWorkloads, err := o.listInitializedLocalWorkloads()
+			if err != nil {
+				return nil, err
+			}
+			workloadsToAppend := selector.FilterOutItems(initializedWorkloads, specifiedWorkloadList, func(s string) string { return s })
+			if len(workloadsToAppend) != 0 {
+				groupsMap[-1] = append(groupsMap[-1], workloadsToAppend...)
+			}
+		} else {
+			// otherwise, add all unspecified local workloads.
+			localWorkloads, err := o.listLocalWorkloads()
+			if err != nil {
+				return nil, err
+			}
+			workloadsToAppend := selector.FilterOutItems(localWorkloads, specifiedWorkloadList, func(s string) string { return s })
+			if len(workloadsToAppend) != 0 {
+				groupsMap[-1] = append(groupsMap[-1], workloadsToAppend...)
+			}
+		}
+	}
+
+	type workloadPriority struct {
+		priority  int
+		workloads []string
+	}
+	deploymentGroups := make([]workloadPriority, len(groupsMap))
+	i := 0
+	for k, v := range groupsMap {
+		deploymentGroups[i] = workloadPriority{
+			priority:  k,
+			workloads: v,
+		}
+		i++
+	}
+	// sort by priority
+	sort.Slice(deploymentGroups, func(i, j int) bool { return deploymentGroups[i].priority < deploymentGroups[j].priority })
+
+	res := make([][]string, len(deploymentGroups))
+	for i, g := range deploymentGroups {
+		res[i] = g.workloads
+	}
+	if deploymentGroups[0].priority != -1 {
+		return res, nil
+	}
+
+	// rotate the array to the left one element so that the -1 priority appears at the end.
+	first := res[0]
+	for i := 1; i < len(res); i++ {
+		res[i-1] = res[i]
+	}
+	res[len(res)-1] = first
+
+	return res, nil
+}
+
+func (o *deployOpts) listStoreWorkloads() ([]*config.Workload, error) {
+	if o.storeWorkloads == nil {
+		wls, err := o.store.ListWorkloads(o.appName)
+		if err != nil {
+			return nil, fmt.Errorf("retrieve store workloads: %w", err)
+		}
+		o.storeWorkloads = wls
+	}
+	return o.storeWorkloads, nil
+}
+
+func (o *deployOpts) listLocalWorkloads() ([]string, error) {
+	if o.wsWorkloads == nil {
+		localWorkloads, err := o.ws.ListWorkloads()
+		if err != nil {
+			return nil, fmt.Errorf("retrieve workspace workloads: %w", err)
+		}
+		o.wsWorkloads = localWorkloads
+	}
+
+	return o.wsWorkloads, nil
+}
+
+func (o *deployOpts) listInitializedLocalWorkloads() ([]string, error) {
+	if o.initializedWsWorkloads == nil {
+		storeWls, err := o.listStoreWorkloads()
+		if err != nil {
+			return nil, err
+		}
+		localWorkloads, err := o.listLocalWorkloads()
+		if err != nil {
+			return nil, err
+		}
+
+		o.initializedWsWorkloads = selector.FilterItemsByStrings(localWorkloads, storeWls, func(workload *config.Workload) string { return workload.Name })
+	}
+	return o.initializedWsWorkloads, nil
+}
+
+type workloadCommand struct {
+	name string
+	cmd  actionCommand
+}
```

Comment:
```suggestion
type workloadCommand struct {
        actionCommand
	name string
}
```
So that we don't have `cmd.cmd` 😂